aten/src/ATen/native/cuda/MemoryAccess.cuh

```diff
@@ -495,7 +497,7 @@ inline C10_HOST_DEVICE int can_vectorize_up_to(const char *pointer) {
   } else if (type_size <= 2 && (address % vec8_alignment == 0)) {
     return 8;
   } else
-#else
+#elif defined(CUDA_VERSION) && CUDA_VERSION >= 12080
```

Comment:
Shouldn't there be some logic to handle the case when CUDA_VERSION < 12080?

Reply:
Hi @ZainRizvi this is basically redoing https://github.com/pytorch/pytorch/pull/145746  only if CUDA >= 12.8

Reply:
Hence this code should not be applied by default but only for CUDA 12.8 
```
if (address % vec8_alignment == 0) {
   return 8;
  } else
```